### ISSUE TITLE
colorscheme resets background to light; setting background=dark will reset the colorscheme

### ISSUE THREAD

**Issue by mission712** [neovim/neovim#8595]
- `nvim --version`: NVIM v0.3.0
- Vim (version: 8.0 Included patches: 1-1766) behaves differently? yes
- Operating system/version: debian sid
- Terminal name/version: xfce4-terminal 0.8.7.4
- `$TERM`: xterm-256colors

### Steps to reproduce using `nvim -u NORC`

Requires a vimrc loading the packages vim-airline/vim-airline and crusoexia/vim-monokai

### Actual behaviour

When I do `:color monokai`, the background variable is set to `light` (nothing in monokai.vim sets this). If I do `:set background=dark` however, the color scheme is reset to `default`. When `color monokai` is set, airline turns white because the background variable changes. I am reporting here because this seems to be an error related to the `termguicolors` command or something like that, since the following makes it work:

```vimscript
color monokai
set termguicolors background=dark
set notermguicolors
let g:airline_theme = 'minimalist'
```
This makes the background color stay at dark while the monokai colorscheme is still selected, allowing normal operation. I did not observe this issue with nvim 0.2.3. 

### Expected behaviour

background variable to stay at dark while changing colorschemes.

**Comment by justinmk** [comment]
> nothing in monokai.vim sets this

Actually, it does. monokai [sets ctermfg=234](https://github.com/crusoexia/vim-monokai/blob/f55dfa5100a1bde99f67c85db42172ec4f5ef066/colors/monokai.vim#L74) for  the `Normal` hl group, which Vim (and Nvim) infers as a "light" background.

> this seems to be an error related to the termguicolors command or something like that, since the following makes it work:

termguicolors uses the gui colors from monokai, so maybe it's skipping the "light background detection" mentioned above.

I think this "auto detection" is just a source of pain, I've never known a case where it actually helps anyone.

**Comment by mission712** [comment]
Hmmm, I see. What about `set background=dark` resetting the color scheme, though? I didn't have this back in neovim 0.2.3, so something must have regressed.

**Comment by justinmk** [comment]
I can't imagine why that would have changed since 0.2.3. Can you bisect to the exact commit that caused it?

**Comment by justinmk** [comment]
does https://github.com/neovim/neovim/pull/8597 fix the issue for you?

**Comment by mission712** [comment]
Should I try building your neovim and see if it works now?

**Comment by justinmk** [comment]
yes

**Comment by justinmk** [comment]
yes, from https://github.com/neovim/neovim/pull/8597 or this branch: https://github.com/justinmk/neovim/tree/hl-do-not-change-bg

**Comment by mission712** [comment]
I'm getting a bunch of these errors:
```
/home/m712/neovim/src/nvim/tui/tui.c:62:43: error: ‘unibi_var_t {aka struct <anonymous>}’ has no member named ‘i’; did you mean ‘i_’?
 #define UNIBI_SET_NUM_VAR(var, num) (var).i = (num);
```
when building from the hl-do-not-change-bg branch.

**Comment by justinmk** [comment]
@mission712 Just to be clear, you mean that https://github.com/neovim/neovim/pull/8597 fixes the issue? usually we don't close issues until the PR is merged :)

**Comment by mission712** [comment]
Well, I compiled your branch which merges using that PR, but yes, #8597 fixes the issue. I'll reopen until it's merged, though.

**Comment by justinmk** [comment]
@mission712 #8606 was merged (which is different than  #8597, the PR you tried). If you rebuild Nvim HEAD, is your issue resolved? Or is #8597 still needed?

**Comment by mission712** [comment]
Sorry for the late reply. I just built neovim HEAD and it seems like the issue is gone. I don't think #8597 is needed at this point.
